src/interfaces/amalgamation-i.ts

```diff
@@ -0,0 +1,66 @@
+import { CorpTypeCd, AmalgamationTypes, CorrectNameOptions, FilingTypes } from '@bcrs-shared-components/enums'
+import type {
+  CompletingPartyIF, ContactPointIF, CourtOrderIF, NameTranslationIF, ShareClassIF
+} from '@bcrs-shared-components/interfaces'
+
+/**
+ * A filing's amalgamation application object from the API. See:
+ * https://github.com/bcgov/business-schemas/blob/main/src/registry_schemas/schemas/amalgamation_application.json
+ */
+export interface RegisteredRecordsAddressesI {
+  registeredOffice: deliveryAndMailingAddressI
+  recordsOffice?: deliveryAndMailingAddressI
+}
+
+export interface NameRequestFilingI {
+  legalType: CorpTypeCd
+  legalName?: string
+  nrNumber?: string
+  correctNameOption?: CorrectNameOptions
+}
+
+export interface AmalgamationApplicationI {
+  amalgamatingBusinesses: any[]
+  courtApproval: boolean
+  type: AmalgamationTypes
+  nameRequest: NameRequestFilingI
+  nameTranslations: NameTranslationIF[]
+  offices: RegisteredRecordsAddressesI | object
+  contactPoint: ContactPointIF
+  parties: CompletingPartyIF[]
+
+  // BEN / CC / BC / ULC only:
+  shareStructure?: {
+    shareClasses: ShareClassIF[]
+  }
+  incorporationAgreement?: {
+    agreementType: string
+  }
+  // ULC only:
+  courtOrder?: CourtOrderIF
+}
+
+/** Amalgamation Application filing interface. */
+export interface AmalgamationApplicationFilingI {
+  header: {
```

Comment:
can you add a 'FilingI' with the header + business sections (these are shared across all filings) and then extend it in the new filing interfaces?